### PR TITLE
[FIX] web_editor: allow to reset background color

### PR DESCRIPTION
**Pull request by dhba-odoo**
**Current behavior before PR:**

When copy-pasting from Discord into Notes, get a text with color and background
color. But when clicking on the trash icon inside the toolbar to remove the
background color, it doesn't remove the background color.

**Desired behavior after PR is merged:**

Allowed to remove background color from text using trash icon.

Task-2889682

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 1]
```diff
@@ -2242,6 +2242,9 @@ export class OdooEditor extends EventTarget {
      * @param {Node} node
      */
     _cleanForPaste(node) {
+        if (node.nodeName === 'FONT') {
+            return;
+        }
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2247]: I'm very confused. `FONT` tags are already allowed. Why is this needed ? And with this return here, why do we still need to add more code below if the function returns directly for `FONT` tags ?

Comment (dhba-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2247]: Hello @dmo-odoo 
when we copy-paste from discord. the copied text that we paste in the editor comes in a span tag with color style instead of a font tag.
so in my changes, I have converted that span tag into a font tag.

if I don't return the font tag here then it will be going into an infinite loop because of the else if part from where _cleanForPaste calls again.

Comment (dmo-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2247]: I don't think this is the right fix. I think it would be better to make a change in the function that actually sets the background color to make it recognize span tags as well as font tags.

Comment (dhba-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js]: Done.

File: addons/web_editor/static/lib/odoo-editor/src/commands/commands.js [hunk 2]
```diff
@@ -551,9 +551,9 @@ export const editorCommands = {
         // Get the <font> nodes to color
         const selectedNodes = getSelectedNodes(editor.editable);
         const fonts = selectedNodes.flatMap(node => {
-            let font = closestElement(node, 'font');
+            let font = closestElement(node, 'font') ? closestElement(node, 'font') : closestElement(node, 'span');
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/commands/commands.js:554]: IIRC `closestElement` takes a selector as argument so giving it `"font, span[style]"` as argument should suffice and simplify this line. Tell me if that doesn't work.

Comment (dhba-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js]: @dmo-odoo Yes, it's working.

Comment (dhba-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js]: Hello @dmo-odoo , I have come across one problem with the last fix in this PR.
if we select like `<p> [ a <span font-size: 8px> b </span> c ]</p>` and apply color at that time result that I am getting is `<p> [ <font style="color: rgb(255, 0, 0)"> a <span font-size: 8px> <font style="color: rgb(255, 0, 0)"> b </font></span> c </font>]</p>`.
now if we want to remove color for only text that is contained in the span we will not able to remove it.

`let font = closestElement(node, 'font') ? closestElement(node, 'font') : closestElement(node, 'span');`
I have checked the same with the above code and I am not getting issues because here I am getting a result `<p> [ <font style="color: rgb(255, 0, 0)"> a <span font-size: 8px>  b </span> c </font>]</p>`

Comment (dmo-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js]: I see, thank you. Does it work if you use this instead:
```js
let font = closestElement(node, 'font') || closestElement(node, 'span');
```
If it works then let's do that.

Comment (dhba-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js]: @dmo-odoo Thanks for the suggestion. it's working properly.

**Comment by dmo-odoo** [comment]
@robodoo r+

**Comment by dmo-odoo** [comment]
@dhba-odoo Thank you !
